### PR TITLE
Remove deprecated \Doctrine\DBAL\Connection class

### LINKED ISSUE #12466: Remove Pimcore\Db\ConnectionInterface and Pimcore\Db\PimcoreExtensionstrait
### Improvement description

See #12123 
remove traits, interfaces and BC layer (union type hints, ...)

### PR DESCRIPTION
**Pull request by blankse**
Followup to https://github.com/pimcore/pimcore/issues/12363

### CODE REVIEW

**Comment by blankse** [comment]
See also https://github.com/pimcore/pimcore/pull/12919 for 10.5 branch

File: doc/Development_Documentation/26_Best_Practice/55_Primary-Replica_Database_Connection.md [hunk 1]
```diff
@@ -1,61 +0,0 @@
-
-# How to configure Pimcore to use a Primary/Replica Database Connection 
-**IMPORTANT**: Please be aware that the primary/replica connection can only be used for a clustered MariaDB/MySQL environment, **NOT**
```
Comment (brusch) [inline doc/Development_Documentation/26_Best_Practice/55_Primary-Replica_Database_Connection.md:3]: This is still relevant information. 
I'd keep that page and for configuration I'd link to the official primary/replica docs of Doctrine Bundle. WDYT?

**Review by brusch — CHANGES_REQUESTED** [review]

**Comment by brusch** [comment]
Related to #10613

**Comment by brusch** [comment]
@blankse as https://github.com/pimcore/pimcore/pull/12919 got merged, this PR causes conflicts, can you please check & fix them - thanks a lot 😊

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -10,7 +10,8 @@
 - [Listings] Removed `JsonListing`, please see [#12877](https://github.com/pimcore/pimcore/pull/12877) for details.
 - [Traits] The traits PackageVersionTrait and StateHelperTrait in lib/Extension/Bundle/Traits have been marked as
    internal, please see [#12757](https://github.com/pimcore/pimcore/pull/12757) for details.
-
+- Removed deprecated `Pimcore\Db\ConnectionInterface` interface, `Pimcore\Db\Connection` class and `Pimcore\Db\PimcoreExtensionsTrait` trait.
+  Column identifiers for the `insert()` method data must be self quoted now. You can use the `Pimcore\Db\Helper::quoteDataIdentifiers()` method for that.
```
Comment (brusch) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
  Column identifiers for the `insert()` and `update()` method data must be self quoted now. You can use the `Pimcore\Db\Helper::quoteDataIdentifiers()` method for that.
```

**Comment by blankse** [comment]
@brusch I merged 10.5 branch and resolve the conflicts. I also added the replica docs again. Can you test it if it works?

**Comment by brusch** [comment]
@blankse can you merge `11.x` branch again? Thanks!

**Comment by blankse** [comment]
@brusch Done

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md (hunk 2 above)
Comment (brusch) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: @blankse I'm afraid that this change (no auto quoting anymore of column names) has some bigger impact and needs some more attention, e.g. here: 

https://github.com/pimcore/pimcore/blob/cd48e25f5c80052e73cc378d77b18effba5c111f/models/Document/Dao.php#L84

**Review by brusch — CHANGES_REQUESTED** [review]

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md (hunk 2 above)
Comment (blankse) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: @brusch This insert is already customized. But I'm looking for other insert/update methods.

Comment (brusch) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: @blankse cool, thanks! 😊

**Review by aashan10 — APPROVED** [review]

**Comment by brusch** [comment]
Thank you very much 👍

**Pull request merged**